examples/pose_landmarker/ios/PoseLandmarker/Services/CameraFeedService.swift

```diff
@@ -0,0 +1,398 @@
+    NotificationCenter.default.addObserver(self, selector: #selector(CameraFeedService.sessionRuntimeErrorOccured(notification:)), name: NSNotification.Name.AVCaptureSessionRuntimeError, object: session)
+    NotificationCenter.default.addObserver(self, selector: #selector(CameraFeedService.sessionWasInterrupted(notification:)), name: NSNotification.Name.AVCaptureSessionWasInterrupted, object: session)
+    NotificationCenter.default.addObserver(self, selector: #selector(CameraFeedService.sessionInterruptionEnded), name: NSNotification.Name.AVCaptureSessionInterruptionEnded, object: session)
+  }
+
+  private func removeObservers() {
+    NotificationCenter.default.removeObserver(self, name: NSNotification.Name.AVCaptureSessionRuntimeError, object: session)
+    NotificationCenter.default.removeObserver(self, name: NSNotification.Name.AVCaptureSessionWasInterrupted, object: session)
+    NotificationCenter.default.removeObserver(self, name: NSNotification.Name.AVCaptureSessionInterruptionEnded, object: session)
+  }
+
+  // MARK: Notification Observers
+  @objc func sessionWasInterrupted(notification: Notification) {
+
+    if let userInfoValue = notification.userInfo?[AVCaptureSessionInterruptionReasonKey] as AnyObject?,
+       let reasonIntegerValue = userInfoValue.integerValue,
+       let reason = AVCaptureSession.InterruptionReason(rawValue: reasonIntegerValue) {
+      print("Capture session was interrupted with reason \(reason)")
+
+      var canResumeManually = false
+      if reason == .videoDeviceInUseByAnotherClient {
+        canResumeManually = true
+      } else if reason == .videoDeviceNotAvailableWithMultipleForegroundApps {
+        canResumeManually = false
+      }
+
+      self.delegate?.sessionWasInterrupted(canResumeManually: canResumeManually)
+
+    }
+  }
+
+  @objc func sessionInterruptionEnded(notification: Notification) {
+    self.delegate?.sessionInterruptionEnded()
+  }
+
+  @objc func sessionRuntimeErrorOccured(notification: Notification) {
+    guard let error = notification.userInfo?[AVCaptureSessionErrorKey] as? AVError else {
+      return
+    }
+
+    print("Capture session runtime error: \(error)")
+
+    guard error.code == .mediaServicesWereReset else {
+      self.delegate?.didEncounterSessionRuntimeError()
+      return
+    }
+
+    sessionQueue.async {
+      if self.isSessionRunning {
+        self.startSession()
+      } else {
+        DispatchQueue.main.async {
+          self.delegate?.didEncounterSessionRuntimeError()
+        }
+      }
+    }
+  }
+}
+
+/**
+ AVCaptureVideoDataOutputSampleBufferDelegate
+ */
+extension CameraFeedService: AVCaptureVideoDataOutputSampleBufferDelegate {
+
+  /** This method delegates the CVPixelBuffer of the frame seen by the camera currently.
+   */
+  func captureOutput(_ output: AVCaptureOutput, didOutput sampleBuffer: CMSampleBuffer, from connection: AVCaptureConnection) {
+      let imageBuffer = CMSampleBufferGetImageBuffer(sampleBuffer)!
```

Comment:
All of this seems to be unused since we are passing the sample buffer with the device orientation for the calculations. Can you remove this?

Reply:
Please disregard this comment. I got confused. This is correct. Can you replace the force unwrap with a guard condition?